spark/src/test/scala/org/apache/comet/CometCastSuite.scala

```diff
@@ -322,11 +322,16 @@ class CometCastSuite extends CometTestBase with AdaptiveSparkPlanHelper {
     castTest(generateInts(), DataTypes.DoubleType)
   }
 
-  ignore("cast IntegerType to DecimalType(10,2)") {
-    // Comet should have failed with [NUMERIC_VALUE_OUT_OF_RANGE] -1117686336 cannot be represented as Decimal(10, 2)
+  test("cast IntegerType to DecimalType(10,2)") {
     castTest(generateInts(), DataTypes.createDecimalType(10, 2))
   }
 
+  test("cast IntegerType to DecimalType(10,2) overflow check") {
+    val intToDecimal10OverflowValues =
+      Seq(Int.MinValue, -100000000, -100000001, 100000000, 100000001, Int.MaxValue).toDF("a")
+    castTest(intToDecimal10OverflowValues, DataTypes.createDecimalType(10, 2))
+  }
```

Comment:
This test does not test all of the native implementation code. For example, it did not test the case where the `checked_mul` returned `None`. I suggest testing with other precisions and scales, including `DecimalType.MAX_PRECISION` and `DecimalType.MAX_SCALE`.